.circleci/config.yml

```diff
@@ -24,21 +35,33 @@ commands:
             BASE_COMMIT=$(git merge-base HEAD remotes/origin/"${ROOT_BRANCH}")
             echo "Checking for changes in branch ${CIRCLE_BRANCH}"
             echo " - Excluding << parameters.exclude >>"
+            echo " - Requiring << parameters.require >>"
+            echo " - Depth << parameters.search_depth >>"
             echo " - Git repository root at ${CIRCLE_WORKING_DIRECTORY}"
             echo " - Base ${ROOT_BRANCH} commit ${BASE_COMMIT}"
+
+            checkModified () {
+              file=$1
+
+              BASE_FILE_HASH=$(git log -1 "${BASE_COMMIT}" --format=format:%H --full-diff "${file}")
+              CURRENT_FILE_HASH=$(git log -1 --format=format:%H --full-diff "${file}")
+              if [ ! $BASE_FILE_HASH = $CURRENT_FILE_HASH ]; then
+                echo " --- ▶️ ${file}/ has been modified on this branch. Continuing the job..."
+                set +e
+                # exit 0
+              fi
+            }
+
             # Test all top-level directories
-            for file in $PWD/*/
-            do
-              FILE_NAME=$(basename $file)
-              if [[ ! " << parameters.exclude >> " =~ " ${FILE_NAME} " ]]; then
-                echo " -- Checking ${FILE_NAME}"
-                BASE_FILE_HASH=$(git log -1 "${BASE_COMMIT}" --format=format:%H --full-diff "${file}")
-                CURRENT_FILE_HASH=$(git log -1 --format=format:%H --full-diff "${file}")
-                if [ ! $BASE_FILE_HASH = $CURRENT_FILE_HASH ]; then
-                  echo " --- ▶️ ${FILE_NAME}/ has been modified on this branch. Continuing the job..."
-                  set +e
-                  exit 0
-                fi
+            for file in $(find $PWD/* -maxdepth << parameters.search_depth >> -not -path '*/\.*'); do
```

Comment:
I think we do want to include hidden files -- if .eslintrc.js changes, we need to run the linter tests.